session_rpcserver.go

```diff
@@ -214,15 +215,92 @@ func (s *sessionRpcServer) resumeSession(sess *session.Session) error {
 		return nil
 	}
 
-	authData := []byte(fmt.Sprintf("%s: %s", HeaderMacaroon, mac))
+	var (
+		isFirstConn             = sess.RemotePublicKey == nil
+		onNewStatus             func(s mailbox.ServerConnStatus)
+		firstConnectionDeadline *time.Timer
+		firstConnTimout         = make(chan struct{})
+		exitFirstConnDeadline   = make(chan struct{})
+	)
+
+	// If this is the first time the session is being spun up then we will
+	// kick off a timer to revoke the session after a timeout unless an
+	// initial connection is made. We identify such a session as one that
+	// we do not yet have a static remote pub key for.
+	if isFirstConn {
+		deadlineDuration := sess.CreatedAt.Add(
+			s.cfg.firstConnectionDeadline,
+		)
+
+		if deadlineDuration.Before(time.Now()) {
+			log.Debugf("Deadline for session %s has already "+
+				"passed. Revoking session", pubKeyBytes)
+
+			err = s.db.RevokeSession(pubKey)
+			if err != nil {
+				log.Debugf("error revoking session: "+"%v", err)
```

Comment:
Any reason for not returning the error here? Or was this intended to be a `return fmt.Errorf()`? Also, string concatenation artifacts.

Reply:
erg yeah sorry 🙈 also, there should be a `return` after this 🙈